### PR TITLE
Deploy swck snapshot charts

### PR DESCRIPTION
**Pull request by kezhenxu94**

### CODE REVIEW

**Comment by wu-sheng** [comment]
Why is the helm going to have snapshot now? I don't think we have changes for this frequently.

**Comment by kezhenxu94** [comment]
> Why is the helm going to have snapshot now? I don't think we have changes for this frequently.

- I plan to migrate our showcase repo to Helm Chart, where we will use snapshot Chart in this repo.
- I plan to also set up workflow to publish the Helm Chart to Docker, which will leverage the same workflow in this patch.

File: chart/operator/Chart.yaml [hunk 1]
```diff
@@ -14,7 +14,7 @@
 # limitations under the License.
 
 apiVersion: v2
-name: skywalking-swck-operator
+name: skywalking-helm-swck-operator
```
Comment (kezhenxu94) [inline chart/operator/Chart.yaml:17]: We will publish these to hub.docker.io, and to make it clear this is a Helm Chart, not a Docker image, I change the names to be prefixed with `skywalking-helm-*`, and will ask infra team to create the repositories with the same name prefix

Comment (wu-sheng) [reply chart/operator/Chart.yaml:17]: So, this is a helm for swck?

Comment (kezhenxu94) [reply chart/operator/Chart.yaml:17]: > So, this is a helm for swck?

Yes, as the title indicates. swck contains two Helm Charts actually, namely, operator and adapter

File: .github/workflows/publish-helm.yaml [hunk 2]
```diff
@@ -23,6 +23,7 @@ on:
 
 env:
   HUB: ghcr.io/apache/skywalking-kubernetes
+  VERSION: 0.0.0-${{ github.sha }}
```
Comment (wu-sheng) [inline .github/workflows/publish-helm.yaml:26]: As a snapshot, could we have a version prefix?

Comment (kezhenxu94) [reply .github/workflows/publish-helm.yaml:26]: > As a snapshot, could we have a version prefix?

like this `-snapshot`?

Comment (wu-sheng) [reply .github/workflows/publish-helm.yaml:26]: prefix, I mean, such as 4.5.0-xxxx

Comment (kezhenxu94) [reply .github/workflows/publish-helm.yaml:26]: > prefix, I mean, such as 4.5.0-xxxx

Sure. the rationale here I use 0.0.0-xxx is that, to mimic the same version style as our existing snapshot Docker images, whose version is simply `<sha>`, without a version prefix like `9.5.0-<sha>`, when we want to use a snapshot of a particular commit in the main branch, we don't have to find out what version is it currently, we can just copy the commit sha and use it `0.0.0-<sha>`.


> The `0.0.0-` prefix is just to adhere the version format of Helm Chart, it must be `\d+.\d+.\d-\w+` style.

**Comment by innerpeacez** [comment]
> > Why is the helm going to have snapshot now? I don't think we have changes for this frequently.
> 
> * I plan to migrate our showcase repo to Helm Chart, where we will use snapshot Chart in this repo.

I actually have this idea, when I use otel.

File: chart/operator/Chart.yaml (hunk 1 above)
Comment (innerpeacez) [reply chart/operator/Chart.yaml:17]: Can't dockerhub directly know that this is a chart type?

Comment (kezhenxu94) [reply chart/operator/Chart.yaml:17]: > Can't dockerhub directly know that this is a chart type?

Of course it knows, I meant I want to make it clear for our users.

**Review by innerpeacez — APPROVED** [review]

File: chart/operator/Chart.yaml (hunk 1 above)
Comment (wu-sheng) [reply chart/operator/Chart.yaml:17]: @innerpeacez I think here is about avoiding confusion of another existing repo, https://hub.docker.com/r/apache/skywalking-swck

Comment (innerpeacez) [reply chart/operator/Chart.yaml:17]: got it.

**Pull request merged**